agent/src/analysis/pipeline_analyzer.py

```
@@ -121,6 +124,69 @@ def add_metadata_files(
+    # Single path for this termination — no ambiguity
+    if len(term_paths) == 1:
+        return term_paths[0][0]  # node_ids tuple
+
+    # Multiple paths — check cache first
+    cache_key = (tuple(sorted(specific_image.properties)), specific_image.suffix, term_type)
+    if cache_key in path_cache:
+        return path_cache[cache_key]
+
+    # Match by comparing expected files against each candidate path
+    base = f"{specific_image.camera_id}{specific_image.counter}"
+    match_expected_set = set(f.lower() for f in best_match.expected_files)
```

Ruff (0.15.0)

[convention] Unnecessary generator (rewrite as a set comprehension)

Rewrite as a set comprehension

(C401)